it/uk/gov/hmrc/helptosave/MissingMandatoryData.scala

```diff
@@ -0,0 +1,175 @@
+package uk.gov.hmrc.helptosave
+
+import java.net.URLEncoder
+import java.time.LocalDate
+import java.util.Base64
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Matchers, WordSpec}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatestplus.play.guice.GuiceOneServerPerSuite
+import play.api.http.Status
+import play.api.http.Status._
+import play.api.libs.json.Json
+import play.api.libs.ws.{WSClient, WSResponse}
+import uk.gov.hmrc.helptosave.connectors.CitizenDetailsConnector.{CitizenDetailsAddress, CitizenDetailsPerson, CitizenDetailsResponse}
+import uk.gov.hmrc.helptosave.connectors.UserDetailsConnector.UserDetailsResponse
+import uk.gov.hmrc.helptosave.models.MissingUserInfo.{Email, Surname}
+import uk.gov.hmrc.helptosave.models.{Address, UserInfo}
+import uk.gov.hmrc.helptosave.services.UserInfoService.UserInfoServiceError.MissingUserInfos
+import uk.gov.hmrc.helptosave.support.FakeRelationshipService
+import uk.gov.hmrc.helptosave.util.NINO
+
+
+class MissingMandatoryData extends WordSpec
+  with Matchers
+  with ScalaFutures
+  with FakeRelationshipService
+  with BeforeAndAfterAll
+  with BeforeAndAfterEach
+ with GuiceOneServerPerSuite {
+
+  implicit override val patienceConfig = PatienceConfig(timeout = Span(5, Seconds), interval = Span(100, Millis))
+
+
+  override lazy val port: Int = 7001
+
+  val wsClient = app.injector.instanceOf[WSClient]
+
+  val url = "http://localhost:7001/help-to-save/eligibility-check"
+
+  def decode(encodedNINO: String): NINO = {
+    new String(Base64.getDecoder.decode(encodedNINO))
+  }
+
+  def checkEligibility(encodedNino: String, userDetailsURI: String): WSResponse = {
+    wsClient
+      .url(s"$url?nino=${decode(encodedNino)}&userDetailsURI=${URLEncoder.encode(userDetailsURI)}")
+      .get()
+      .futureValue
+  }
+
+  def mockEligibilityCheck(result: Boolean) =
+    wireMockServer.stubFor(
+      get(urlPathMatching("/help-to-save-stub/eligibilitycheck.*"))
+        .willReturn(
+          aResponse()
+            .withStatus(Status.OK)
+            .withBody(s"""{ "isEligible" : $result }""")
+        )
+    )
+
+  def mockUserDetailsCall(userDetails: UserDetailsResponse): Unit = {
+    val body = Json.toJson(userDetails).toString
+
+    wireMockServer.stubFor(
+      get(urlPathMatching("/user-details.*"))
+        .willReturn(
+          aResponse()
+            .withStatus(Status.OK)
+            .withBody(body)
+        )
+    )
+  }
+
+  def mockCitizenDetailsCall(citizenDetails: CitizenDetailsResponse): Unit = {
+    val body = Json.toJson(citizenDetails).toString
+    wireMockServer.stubFor(
+      get(urlPathMatching(".*/citizen-details.*"))
+        .willReturn(
+          aResponse()
+            .withStatus(Status.OK)
+            .withBody(body)
+        )
+    )
+  }
```

Comment:
having to pass in the url each time each time you want to stub out might be annoying though - probably something like this would avoid code duplication:
```scala
def stub(urlRegex: String, body: String): Unit =
  wireMockServer.stubFor(
       get(urlPathMatching(urlRegex))
         .willReturn(
           aResponse()
             .withStatus(Status.OK)
             .withBody(body)
         )
     )

def mockCitizenDetailsCall(citizenDetails: CitizenDetailsResponse): Unit =
stub(".*/citizen-details.*", Json.toJson(citizenDetails).toString)
```

etc.